banzai/utils/file_utils.py

```diff
@@ -16,6 +23,46 @@ def post_to_archive_queue(image_path, broker_url, exchange_name='fits_files'):
         producer.release()
 
 
+def post_to_ingester(filepath):
+    retry = True
+    try_counter = 1
+    ingester_response = {}
+    with open(filepath, 'rb') as f:
+        while retry:
+            try:
+                ingester_response = ingester.upload_file_and_ingest_to_archive(f)
+                retry = False
+            except DoNotRetryError as exc:
+                logger.warning('Exception occured: {0}. Aborting.'.format(exc),
+                               extra_tags={'filename': filepath})
+                retry = False
+            except NonFatalDoNotRetryError as exc:
+                logger.debug('Non-fatal Exception occured: {0}. Aborting.'.format(exc),
+                             extra_tags={'filename': filepath})
+                retry = False
+            except RetryError as exc:
+                logger.debug('Retry Exception occured: {0}. Retrying.'.format(exc),
+                             extra_tags={'tags': {'filename': filepath}})
+                retry = True
+                try_counter += 1
+            except BackoffRetryError as exc:
+                logger.debug('BackoffRetry Exception occured: {0}. Retrying.'.format(exc),
+                             extra_tags={'filename': filepath})
+                if try_counter > 5:
+                    logger.warning('Giving up because we tried too many times.', extra_tags={'filename': filepath})
+                    retry = False
+                else:
+                    sleep(5 ** try_counter)
```

Comment:
Is it possible to use celery or something for this? That way you could schedule the task to execute later instead of blocking. https://github.com/LCOGT/ingester/blob/6a4eb8f4de65566a931eecaff84f43e8fca46346/tasks.py#L61

Reply:
This could definitely be a celery task. Good call.